modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java

```diff
@@ -149,7 +149,34 @@ public void insertDatasetDetails(String name, String tenantID, String username,
     }
 
     @Override
-    public void insertDatasetVersionDetails(String datasetId, String tenantId, String version) throws DatabaseHandlerException {
+    public long getDatasetId(String datasetName, int tenantId) throws DatabaseHandlerException {
+
+        Connection connection = null;
+        ResultSet result = null;
+        PreparedStatement statement = null;
+        try {
+            connection = dbh.getDataSource().getConnection();
+            statement = connection.prepareStatement(SQLQueries.GET_DATASET_ID);
+            statement.setString(1, datasetName);
+            statement.setInt(2, tenantId);
+            result = statement.executeQuery();
+            if (result.first()) {
+                return result.getLong(1);
+            } else {
+                throw new DatabaseHandlerException(
+                        "No dataset id associated with dataset name: " + datasetName + " and tenant id:" + tenantId);
+            }
+        } catch (SQLException e) {
+            throw new DatabaseHandlerException(
+                    " An error has occurred while extracting dataset name: " + datasetName + " and tenant id:" + tenantId);
```

Comment:
committed in https://github.com/wso2/product-ml/commit/56497b808effb296cc55d6800b2326ce266e6b8a